govuk-one-login/govuk-one-login-frontend
# Set up front end vitals monitoring - basic monitoring

## Description and Context

This sets up the basic scaffolding for the front end vitals monitoring. 
Please use the README.md for instructions on how to initalise.

I have tested this on the demo application.

### Tickets ##

[- [DFC-XXX](https://govukverify.atlassian.net/browse/DFC-XXX)](https://govukverify.atlassian.net/browse/DFC-595)

### Steps to reproduce ###
To test this locally 

Part one
- Check out the branch
- CD into the directory /packages/frontend-vitals-monitoring
- Run `npm build` 
- Copy the entire folder `frontend-vitals-monitoring` and paste into 
<img width="278" alt="Screenshot 2024-06-24 at 15 03 41" src="https://github.com/govuk-one-login/govuk-one-login-frontend/assets/38694448/5f9edd39-dde8-49f8-980e-e3dbc372d0a4">

Part two
Open the demo app, app.js
add in 

`const { frontendVitalsInit } = require("@govuk-one-login/frontend-vitals-monitoring")
frontendVitalsInit(1000)`
to initialise - this will start the application

Successful logging should show 

<img width="525" alt="Screenshot 2024-06-24 at 15 11 36" src="https://github.com/govuk-one-login/govuk-one-login-frontend/assets/38694448/a555cf4e-954b-4514-a3d2-6e63dc9cf1a0">
